File: 942-di-string-match/942-di-string-match.go
func diStringMatch(s string) []int {
  min, max := 0, len(s)
  ans := make([]int, len(s), len(s) + 1)
  for i, v := range s {
    switch v {
      case 'I':
        ans[i] = min
        min++
      case 'D':
        ans[i] = max
        max--
    }
  }
  return append(ans, min)
}

File: 942-di-string-match/942-di-string-match.py
class Solution:
  def diStringMatch(self, s: str) -> List[int]:
    min = 0
    max = len(s)
    ans = []
    for ch in s:
      if ch == 'I':
        ans.append(min)
        min += 1
      else:
        ans.append(max)
        max -= 1
    ans.append(min)
    return ans